wandb/stats.py

```diff
@@ -1,20 +1,39 @@
 import collections
-import os
-from pynvml import *
+import pynvml
 import time
+import os
 from numbers import Number
 import threading
 from wandb import util
 from wandb import termlog
 psutil = util.get_module("psutil")
 
+def gpu_in_use_by_this_process(gpu_handle):
+    our_pids = os.getpid()
+    our_parent_pid = os.getppid()
```

Comment:
Not supported in windows on python 2.7

And in general I am not sure about this approach, what happens if the user is using data parallel pytorch and spinning up multiple processes?

Reply:
Updated to use psutil, and to recursively list all child pids. Still possible edge cases around `wandb run`, as we discussed offline, but much more robust overall.